### ISSUE TITLE
Crash on carriage return after typing (or pasting) a URL

### ISSUE THREAD

**Issue by ShawnDEvans** [giuspen/cherrytree#1369]
**Version, Operative system**
Cherry Tree Version:
0.99.22
OS Version:
Pop!_OS 20.10
Gnome Version: 3.38.1
Uname: Linux pop-os 5.8.0-7630-generic #32~1605108853~20.04~8bcf10e-Ubuntu SMP Wed Nov 11 22:42:16 UTC  x86_64 x86_64 x86_64 GNU/Linux

**Describe the bug**
When typing a URL in the content pane of a node, whenever the enter key is pressed (carriage return) to start a new line the app crashes. 

**To Reproduce**
1. Add a rich text node to any file
2. In the content pane of the node type "http://" (no quotes), and press enter. 
3. App crashes

The same behavior can be observed with any URL, however the bare minimum necessary to crash the app is "http://".

**Comment by txe** [comment]
Hmm, I cannot reproduce this issue. @giuspen , can you?

**Comment by giuspen** [comment]
Not on 20.04, tomorrow I'll test on 20.10 at work where I have a 20.10 VM.
@ShawnDEvans is you have some non-default settings please summarise here.

**Comment by ShawnDEvans** [comment]
I think I have everything dialed to the default settings. It turns out that links for other protocols can also reproduce the crash. So far the following seem to do the trick:

ftp://
http://
https://

If it would be helpful I can upload a redacted config file (/home/username/.config/cherrytree/config.cfg). Let me know!

**Comment by giuspen** [comment]
@ShawnDEvans if you try to reproduce https://github.com/giuspen/cherrytree/issues/1367 that also crashes for you? Could be the same issue with crashes when applying a tag

**Comment by ShawnDEvans** [comment]
@giuspen I was able to reproduce #1367 . Identical crash behavior

**Comment by giuspen** [comment]
OK thanks @ShawnDEvans , I'm going to install 20.10 on a virtual machine, can you confirm that your OS is based on ubuntu 20.10 and you have installed cherrytree via the ppa?

**Comment by ShawnDEvans** [comment]
My pleasure @giuspen, the support is appreciated. I can confirm that Pop_OS is based on Ubuntu 12.10. I installed Cherrytree via the OS package manager, which I think is based upon elementary software center.

**Comment by ciranus** [comment]
> I think is based upon elementary software center

@ShawnDEvans 
The problem with exotic linux distributions, we never know where the packages are coming from.
Can you show this to clarify:

`grep -r '^deb ' /etc/apt/sources{,.list.d/*}.list`

**Comment by ShawnDEvans** [comment]
@ciranus please find below the requested sources:

/etc/apt/sources.list.d/google-chrome.list:deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main
/etc/apt/sources.list.d/metasploit-framework.list:deb http://downloads.metasploit.com/data/releases/metasploit-framework/apt lucid main
/etc/apt/sources.list.d/pop-os-ppa.list:deb http://ppa.launchpad.net/system76/pop/ubuntu groovy main
/etc/apt/sources.list.d/slack.list:deb https://packagecloud.io/slacktechnologies/slack/debian/ jessie main
/etc/apt/sources.list.d/system76-ubuntu-pop-eoan.list:deb http://ppa.launchpad.net/system76/pop/ubuntu focal main

Let me know if you need anything else.

**Comment by ciranus** [comment]
There is no cherrytree package in these repositories.
How did you install cherrytree ?

**Comment by ShawnDEvans** [comment]
I just use the software center via the GUI. I think it came from Flathub/Flatpak. I'd be happy to provide more info, but I'm not terribly familiar with package management on Linux, so I may need some guidance to provide you with more specific details.

**Comment by ciranus** [comment]
Can you try and compare this:

```
wget http://ppa.launchpad.net/giuspen/ppa/ubuntu/pool/main/c/cherrytree/cherrytree_0.99.22-1_amd64.deb
sudo apt install ./cherrytree_0.99.22-1_amd64.deb
```

**Comment by mkamarin** [comment]
I'm experiencing the same issue in Linux Mint 20, under Cinnamon 4.6.7, using Linux kernel 5.4.0-42-generic. I'm using CherryTree 0.99.22 installed via Mint's Software Manager, but it is a Flatpak from Flathub.

It started happening couple of days ago. Before that, it worked fine for several months in the same configuration (with normal software updates). So, a recent system update may have broke it.

I ran it under gdb, but was not able to get a stack trace (catched it too late). In any case, this information may help:
```sh
laptop:~$ flatpak run com.giuspen.cherrytree
[2020-11-25 12:46:08.328] [gtk] [critical] Class::register_derived_type(): base_query.type_name is NULL.
[2020-11-25 12:46:08.333] [   ] [debug] /home/mmarin/.var/app/com.giuspen.cherrytree/config/cherrytree/config.cfg parsed
[2020-11-25 12:46:08.488] [   ] [debug] autosave is started
[2020-11-25 12:46:08.844] [   ] [debug] Node name: NextCloud


laptop:~$ gdb -ex=run --args flatpak run com.giuspen.cherrytree
GNU gdb (Ubuntu 9.2-0ubuntu1~20.04) 9.2
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from flatpak...
(No debugging symbols found in flatpak)
Starting program: /usr/bin/flatpak run com.giuspen.cherrytree
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff43a5700 (LWP 26780)]
[New Thread 0x7ffff3ba4700 (LWP 26781)]
[Detaching after fork from child process 26786]
[Thread 0x7ffff3ba4700 (LWP 26781) exited]
[Thread 0x7ffff43a5700 (LWP 26780) exited]
process 26776 is executing new program: /usr/bin/bwrap
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[Detaching after fork from child process 26790]
[2020-11-25 12:52:31.362] [gtk] [critical] Class::register_derived_type(): base_query.type_name is NULL.
[2020-11-25 12:52:31.367] [   ] [debug] /home/mmarin/.var/app/com.giuspen.cherrytree/config/cherrytree/config.cfg parsed
[2020-11-25 12:52:31.536] [   ] [debug] autosave is started
[2020-11-25 12:52:31.926] [   ] [debug] Node name: NextCloud
[2020-11-25 12:53:48.128] [   ] [debug] Node name: tmp
/usr/include/c++/10.2.0/optional:441: constexpr _Tp& std::_Optional_base_impl<_Tp, _Dp>::_M_get() [with _Tp = Gtk::TextIter; _Dp = std::_Optional_base<Gtk::TextIter, false, false>]: Assertion 'this->_M_is_engaged()' failed.
--Type <RET> for more, q to quit, c to continue without paging--
[Inferior 1 (process 26776) exited with code 0206]
(gdb) bt
No stack.
(gdb) 
```

**Comment by ciranus** [comment]
`/lib/x86_64-linux-gnu/libthread_db.so.1` is from libc6.

Don't know if useful, but which libc6 version ?
`apt list libc6`

**Comment by giuspen** [comment]
I'm reproducing it now, working on it

**Comment by mkamarin** [comment]
```sh
laptop:~$ apt list libc6
Listing... Done
libc6/focal-updates,now 2.31-0ubuntu9.1 amd64 [installed]
libc6/focal-updates 2.31-0ubuntu9.1 i386
```

**Comment by giuspen** [comment]
Should be fixed, as soon as the new flatpak build completes I'll double check (tomorrow morning Europe time), building now on https://flathub.org/builds/#/

**Comment by giuspen** [comment]
Closing as fixed in upcoming 0.99.23
